app/views/foreman/hosts/_registration.html.erb

```diff
@@ -0,0 +1,14 @@
+<% activation_keys = Katello::ActivationKey.authorized(:view_activation_keys).select(:name) %>
+<div class='form-group'>
+
+  <label class='col-md-2 control-label'>
+    <%= _('Activation Key(s)') %>
+    <% help = _('Activation key(s) for Subscription Manager. Multiple keys add separated by comma, example: key1,key2,key3') %>
+    <a rel="popover" data-content="<%= help %>" data-trigger="focus" data-container="body" data-html="true" tabindex="-1">
+      <span class="pficon pficon-info "></span>
+    </a>
+  </label>
+  <div class='col-md-4'>
+    <%= text_field_tag 'activation_key', params[:activation_key], class: 'form-control' %>
```

Comment:
If this is left blank it still sends
```
subscription-manager register --org=Default_Organization --activationkey=""
```

I think "Activation Key" needs to be a required field (if a relevant OS is selected).

The output from my test:
```
[root@katello-client vagrant]# curl -X GET "https://centos7-katello-devel.redhatlaptop.example.com/register?location_id=2&operatingsystem_id=1&organization_id=1" -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjo0LCJpYXQiOjE2MDQ2MDI0MjQsImp0aSI6ImMzMjI3ODJlOTliMzg0M2EwMzY5NTBmYmI3ZDIxNGQzM2E3NmI1N2YyYzg3ZDk5OWE0NTU1MWFiNDA1MWE5YjUiLCJleHAiOjE2MDQ2MTY4MjR9.PF-V4F08dGHdcGzXV3qiqME6s6_28GymAKNOErlpk40' | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 10433    0 10433    0     0  41014      0 --:--:-- --:--:-- --:--:-- 41074
#
# Running registration
#
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9137  100  9137    0     0  1338k      0 --:--:-- --:--:-- --:--:-- 1487k
Loaded plugins: enabled_repos_upload, fastestmirror, package_upload, product-id, search-disabled-repos, subscription-manager

This system is not registered with an entitlement server. You can use subscription-manager to register.

Examining /tmp/tmp.8XjmjPu8RK.rpm: katello-ca-consumer-centos7-katello-devel.redhatlaptop.example.com-1.0-1.noarch
/tmp/tmp.8XjmjPu8RK.rpm: does not update installed package.
Nothing to do
Uploading Enabled Repositories Report
Cannot upload enabled repos report, is this client registered?
Error: Must specify an activation key
This system is not yet registered. Try 'subscription-manager register --help' for more information.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   207    0   164  100    43    912    239 --:--:-- --:--:-- --:--:--   916
ERROR: not_found
Host was not found by the subscription UUID: '', this can happen if the host is registered already, but not to this Foreman
```

Reply:
> If this is left blank it still sends

Yes, but since I can't change template from Katello, I'll do it in this ticket: https://projects.theforeman.org/issues/31240

> I think "Activation Key" needs to be a required field (if a relevant OS is selected).

Yes, I agree, but for now I'd like to keep it as it is and add this validation when we will refactor the form.
Added comment to https://projects.theforeman.org/issues/31240 so we won't forget it.

Reply:
Sounds good to me (as long as one doesn't end up getting released before the other).

In the meantime maybe it's good to add 
```
Required for CentOS and Red Hat Enterprise Linux.
```
to the help text above.  (in other words, remind users that it's needed even before the validation is done)